src/tests/components/CollectionIcons.test.ts

```
@@ -0,0 +1,43 @@
+        const removeButton = getByLabelText("Remove icon Icon 1");
+        await fireEvent.click(removeButton);
+
+        expect(mockOnRemoveIcon).toHaveBeenCalledWith(mockCollection.icons[0], mockCollection);
+    });
+
+    it("calls onSelectUserIcon when an icon is clicked", async () => {
+        const mockOnSelectUserIcon = vi.fn();
+        const mockCollection = {
+            id: "collection-id",
+            icons: [{ id: "icon1", label: "Icon 1", base64Thumbnail: "data:image/png;base64,..." }]
+        } as any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)